crates/primitives/contract/src/class.rs

```diff
@@ -191,19 +189,19 @@ pub mod convert {
             Self {
                 r#type: AbiEventTypeWrapper::from(abi_event_entry.r#type),
                 name: abi_event_entry.name,
-                keys: abi_event_entry.keys.into_iter().map(|v| AbiTypedParameterWrapper::from(v)).collect(),
-                data: abi_event_entry.data.into_iter().map(|v| AbiTypedParameterWrapper::from(v)).collect(),
+                keys: abi_event_entry.keys.into_iter().map(AbiTypedParameterWrapper::from).collect(),
+                data: abi_event_entry.data.into_iter().map(AbiTypedParameterWrapper::from).collect(),
             }
         }
     }
 
-    impl Into<LegacyEventAbiEntry> for AbiEventEntryWrapper {
-        fn into(self) -> LegacyEventAbiEntry {
+    impl From<AbiEventEntryWrapper> for LegacyEventAbiEntry {
+        fn from(abi_event_entry: AbiEventEntryWrapper) -> Self {
             LegacyEventAbiEntry {
-                r#type: self.r#type.into(),
-                name: self.name,
-                keys: self.keys.into_iter().map(|v| v.into()).collect(),
-                data: self.data.into_iter().map(|v| v.into()).collect(),
+                r#type: abi_event_entry.r#type.into(),
+                name: abi_event_entry.name,
+                keys: abi_event_entry.keys.into_iter().map(|v| v.into()).collect(),
```

Comment:
Could you please replace `|v| v.into()` with a reference to the associated `from` function?